sway/desktop/xwayland.c

```diff
@@ -285,6 +306,40 @@ static void handle_request_fullscreen(struct wl_listener *listener, void *data)
 	view_set_fullscreen(view, xsurface->fullscreen);
 }
 
+static void handle_set_title(struct wl_listener *listener, void *data) {
+	struct sway_xwayland_view *xwayland_view =
+		wl_container_of(listener, xwayland_view, set_title);
+	struct sway_view *view = &xwayland_view->view;
+	struct wlr_xwayland_surface *xsurface = view->wlr_xwayland_surface;
+	if (!xsurface) {
```

Comment:
You probably want to check if the xwayland surface is mapped here? xsurface won't be NULL.